tls/s2n_cipher_suites.c

```diff
@@ -131,6 +136,11 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                     continue;
                 }
 
+                /* Don't choose a suite the client shouldn't be able support */
+                if (conn->client_protocol_version < match->minimum_required_tls_version) {
+                    continue;
+                }
+
```

Comment:
Here too, this makes sense, unless it's the last viable option. Do we risk locking some clients out?